randovania/game_description/node_search.py

```diff
@@ -46,10 +45,9 @@ def get_dock_connection_for(n: DockNode):
             new_areas = set()
             for node in area.nodes:
                 connection = None
-                if isinstance(node, DockNode):
+                if isinstance(node, DockNode) and not node.is_teleporter_node():
                     connection = get_dock_connection_for(node).area_identifier
-
-                elif isinstance(node, TeleporterNode) and not ignore_elevators:
+                elif isinstance(node, DockNode) and node.is_teleporter_node() and not ignore_elevators:
```

Comment:
I barely know how this is currently working.
`hint_distributor` is ignoring elevators for the distances.
`hint_formatter` is not ignoring elevators for the distance. 
I don't know how this could be a field of `DockType`, if it sometimes needs to be `false` and sometimes `true`.

Reply:
Make both ignore elevators.

Reply:
...and portals?

Reply:
same, that's the bug it has right now :p

Reply:
https://github.com/randovania/randovania/blob/d9301b98183882ab94b54cac83a150280274153e/randovania/exporter/hints/hint_formatters.py#L57-L63

I did the above. Where the list contains docktype teleporter and portals.
Now, I get some test failures around that. 
Just to make sure: I assume that it is kinda expected because ignoring these two types is an actual logic change. Also the key error with the area means that the area is not reachable anymore without taking an elevator or portal?
If that is right, I could add all areas not present in the result with a distance of 0 (if that means unreachable / infinity) or just do a `get` and return a default of 0.

Reply:
Most of the requested changes are kinda trivial. Would be nice to get some feedback on this one here.

Reply:
Removing teleporter + portals for the hints leads to the test results below.
What I can do about that? See two post above ;)
```
FAILED test/gui/test_game_details_window.py::test_update_layout_description_actual_seed[False] - KeyError: Area[Putrid Alcove]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[1-up to-HintItemPrecision.DETAILED-the Reference Pickup] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[1-up to-HintItemPrecision.PRECISE_CATEGORY-a suit] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[1-up to-HintItemPrecision.BROAD_CATEGORY-a life support system] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[0-up to-HintItemPrecision.DETAILED-the Reference Pickup] - KeyError: Area[Accursed Lake]
FAILED test/gui/test_game_details_window.py::test_update_layout_description_actual_seed[True] - KeyError: Area[Putrid Alcove]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[0-up to-HintItemPrecision.PRECISE_CATEGORY-a suit] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[0-up to-HintItemPrecision.BROAD_CATEGORY-a life support system] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[None-exactly-HintItemPrecision.DETAILED-the Reference Pickup] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[None-exactly-HintItemPrecision.PRECISE_CATEGORY-a suit] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_item[None-exactly-HintItemPrecision.BROAD_CATEGORY-a life support system] - KeyError: Area[Accursed Lake]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_area[2-up to] - KeyError: Area[Great Bridge]
FAILED test/games/prime/patcher_file_lib/test_item_hints.py::test_create_message_for_hint_relative_area[None-exactly] - KeyError: Area[Great Bridge]
FAILED test/games/prime2/patcher/test_patch_data_generator.py::test_generate_patcher_data[False] - KeyError: Area[Putrid Alcove]
FAILED test/games/prime2/patcher/test_patch_data_generator.py::test_generate_patcher_data[True] - KeyError: Area[Putrid Alcove]
```

Reply:
There's two usages of `node_search.distances_to_node`. One is for _placing_ the relative hints, the other is for writing the _text_ of these hints.

Currently when placing we ignore elevators, but not portals. When writing the text, we allow both.
Placing a relative hint can fail (there's rules to avoid "8 rooms away from something" or "1 room away from an expansion"), but naming a hint is always successful as the hint is there and we _need_ to put some text.

Maybe we can refuse to put a hint or something, but that's out of scope for this PR. So the old case of `ignore_elevators==False`, keep it as all dock types allowed.

Reply:
Hmm still don't know if I understood that correctly.
I just set the `dock_types_to_ignore` to be an empty list for the `hint_formatter`. This should be the same as  before with the `ignore_elevators=False`. Placing now ignores portals and elevators and when writing text everything is allowed. Should I add some fixme / todos around that part? Afaik both should ignore elevator and portals such that the text does not contain the distance including elevators and portals.

Could maybe remove the `dock_types_to_ignore` variable in the `hint_formatter`, or using defaults in the `distances_to_node` function again. But that are some nuances.